### PR TITLE
Improve regexp and add two faces

### PR DESCRIPTION
**Pull request by dev-plvlml**
This proposes two changes:
1) Distinguish possible characters for hexadecimal and octal forms, and add support for the unicode form. The full regexp is mostly C/C++ targeted while Ruby and JS become supported as subsets.
2) Add one face for the backslash only and another for a character sequence like it is done for the regexp grouping backslash and a regexp grouping construct ifself in vanilla Emacs.

### CODE REVIEW

**Comment by dgutov** [comment]
Hi! Thanks for the pull request.

But I'd like to be more precise than that. In particular, try to avoid false positives.

For instance, in Ruby the `\U...` escape sequence doesn't do anything. And the `\x...` escape only uses two digits (the rest, if any, are interpreted literally).

Could you split the new `hes-escape-sequence-re` value into two variables (at least)?

If JS escapes are a strict subset of C++ escapes, then I imagine the end result would be one common font-lock keywords list, as well as a separate keywords for C++.

**Comment by dev-plvlml** [comment]
I finally decided to add specific (and full) regexps for all the languages since they all differ here and there. Splitting into distinct (and smaller) variables makes some sense, but can end up in creating distinct versions of `\x`'s, `\u`'s, etc. for most languages.

Also, a new `hes-mode-alist` variable is proposed to handle all the languages in a uniform way from simple ones with the common regexp only to the original `hes-ruby-keywords` way. C/C++ still have some work to do in this direction.

File: highlight-escape-sequences.el [hunk 1]
```diff
@@ -76,17 +182,36 @@ backslash followed by anything else.")
            (set-default symbol value))))
 
 ;;;###autoload
+  (defun turn-on-hes-mode()
```
Comment (dgutov) [inline highlight-escape-sequences.el]: There's extra indent here.

**Comment by dgutov** [comment]
Thanks a lot!

By the way, do you have copyright assignment papers for Emacs signed, or are you willing to sign those? I'd like to put this package into GNU ELPA now.

**Comment by dev-plvlml** [comment]
No, I don't have them signed yet but I'm ready to look into it if it's mandatory.

**Comment by dgutov** [comment]
MELPA has updated automatically already.

Yes, GNU ELPA requires copyright assignment. On the plus side, after that you'll be able to contribute patches to Emacs core as well. The instructions are here: http://git.savannah.gnu.org/cgit/gnulib.git/tree/doc/Copyright/request-assign.future

As the program name (the first question), simply put "Emacs".

**Comment by dgutov** [comment]
@dev-plvlml Any news there?

**Comment by dgutov** [comment]
@dev-plvlml Ping?

**Comment by dev-plvlml** [comment]
@dgutov Awfully sorry, lost in too much work! Gonna do it in a week or so!

**Comment by dgutov** [comment]
@dev-plvlml Cool. Thanks in advance.

**Comment by dgutov** [comment]
Here's our monthly reminder: please do the assignment stuff? :)

**Comment by dgutov** [comment]
@dev-plvlml Any progress?

**Comment by dgutov** [comment]
@dev-plvlml Seriously?

**Comment by dev-plvlml** [comment]
@dgutov The assignment paper is on its way to Boston!

**Comment by dgutov** [comment]
@dev-plvlml That's a great news. Thanks!

**Comment by dev-plvlml** [comment]
@dgutov Signed!

**Comment by dgutov** [comment]
@dev-plvlml Thank you. I've pushed it to GNU ELPA now.

**Pull request merged**